### PR TITLE
adapter oh on V3.5.1

### PR DESCRIPTION
**Pull request by oahcy**
Re: #

### Changelog

* adpter 3.5.1， fix some crash。

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Review by qiuguohua — APPROVED** [review]

File: native/cocos/bindings/jswrapper/napi/Utils.cpp [hunk 1]
```diff
@@ -151,6 +155,13 @@ void seToJsArgs(napi_env env, const ValueArray& args, std::vector<target_value>*
 
 bool setReturnValue(const Value& data, target_value& argv) {
     LOGI("setReturnValue");
+    napi_status status;
```
Comment (minggo) [inline native/cocos/bindings/jswrapper/napi/Utils.cpp]: Can define it in if sentence.

Comment (minggo) [inline native/cocos/bindings/jswrapper/napi/Utils.cpp]: Can define it in if sentence.

File: native/cocos/platform/interfaces/modules/Device.cpp [hunk 2]
```diff
@@ -31,9 +31,28 @@
 #include "platform/interfaces/modules/INetwork.h"
 #include "platform/interfaces/modules/IScreen.h"
 #include "platform/interfaces/modules/IVibrator.h"
+#include "platform/interfaces/modules/ISystemWindow.h"
 
 namespace cc {
 
+int Device::getInnerWidth() {
+    CCASSERT(CC_GET_PLATFORM_INTERFACE(ISystemWindow) != nullptr, "Screen interface does not exist");
```
Comment (minggo) [inline native/cocos/platform/interfaces/modules/Device.cpp]: Please use `CC_ASSERT`

File: native/cocos/platform/interfaces/modules/Device.cpp [hunk 3]
```diff
@@ -31,9 +31,28 @@
...
+    auto viewSize = CC_GET_PLATFORM_INTERFACE(ISystemWindow)->getViewSize();
+    return static_cast<int>(viewSize.x);
+}
+
+int Device::getInnerHeight() {
+    CCASSERT(CC_GET_PLATFORM_INTERFACE(ISystemWindow) != nullptr, "Screen interface does not exist");
```
Comment (minggo) [inline native/cocos/platform/interfaces/modules/Device.cpp]: The same as above.

File: native/cocos/platform/interfaces/modules/Device.cpp [hunk 4]
```diff
@@ -31,9 +31,28 @@
...
+    auto viewSize = CC_GET_PLATFORM_INTERFACE(ISystemWindow)->getViewSize();
+    return static_cast<int>(viewSize.y);
+}
+
+uint32_t Device::getWindowHandler() {
+    CCASSERT(CC_GET_PLATFORM_INTERFACE(ISystemWindow) != nullptr, "Screen interface does not exist");
```
Comment (minggo) [inline native/cocos/platform/interfaces/modules/Device.cpp]: The same as above.

File: native/cocos/platform/interfaces/modules/Device.h [hunk 5]
```diff
@@ -72,6 +72,9 @@ class CC_DLL Device {
         float rotationRateGamma = 0.0F;
     };
 
+    static int getInnerWidth();
+    static int getInnerHeight();
+    static uint32_t getWindowHandler();
```
Comment (minggo) [inline native/cocos/platform/interfaces/modules/Device.h:77]: Is it the pointer of window? If so, then 32 bit is not big enough on 64-bit systems.

File: native/cocos/platform/interfaces/modules/Device.cpp (hunk 4 above)
Comment (oahcy) [reply native/cocos/platform/interfaces/modules/Device.cpp]: this is 3.5, not replace with CC_ASSERT.

Comment (minggo) [reply native/cocos/platform/interfaces/modules/Device.cpp]: It is new codes, please use `CC_ASSERT` as `CCASSERT` is removed in v3.6. And these codes will be merged to develop in future.

File: native/cocos/platform/interfaces/modules/Device.h (hunk 5 above)
Comment (oahcy) [reply native/cocos/platform/interfaces/modules/Device.h:77]: This only considers 32-bit, and it will be changed separately later

Comment (oahcy) [reply native/cocos/platform/interfaces/modules/Device.h:77]: https://github.com/cocos/3d-tasks/issues/13618

**Review by minggo — APPROVED** [review]

**Pull request merged**